crates/router/src/core/admin.rs

```diff
@@ -3199,6 +3312,38 @@ pub async fn delete_connector(
             id: id.clone().get_string_repr().to_string(),
         })?;
 
+    let business_profile = crate::core::utils::validate_and_get_business_profile(
```

Comment:
you can get the business profile directly from the `profile_id` instead of using this function.

Reply:
use this function https://github.com/juspay/hyperswitch/blob/2a790cea3c792586ccba11f3e11fdd9b7a4e02e9/crates/router/src/db/business_profile.rs#L31

Reply:
How though? For `v2` feature flag I don't see a Profile field or any implementation that return Profile

https://github.com/juspay/hyperswitch/blob/64383915bda5693df1cecf6cc5683e8b9aaef99b/crates/hyperswitch_domain_models/src/merchant_account.rs#L168-L181

Reply:
should be done